plugins/modules/fusion_array.py

```diff
@@ -252,17 +255,23 @@ def main():
     array = get_array(module, fusion)
 
     changed = False
+    id = None
+    if array is not None:
+        id = array.id
+
     if not array and state == "present":
         module.fail_on_missing_params(["hardware_type", "host_name", "appliance_id"])
-        changed = create_array(module, fusion) | update_array(
+        changed, id = create_array(module, fusion)
+        update_array(
             module, fusion
         )  # update is run to set properties which cannot be set on creation and instead use defaults
     elif array and state == "present":
-        changed = changed | update_array(module, fusion)
+        changed = update_array(module, fusion)
     elif array and state == "absent":
         changed = changed | delete_array(module, fusion)
-    else:
-        module.exit_json(changed=False)
+
+    if id is not None:
+        module.exit_json(changed=changed, id=id)
```

Comment:
This will return `id` when deleting resources. Not only on update/create. Is that wanted behavior?

Reply:
Why we should return id of deleted resource?

Reply:
right it is bug. Will fix it